src/pages/index.js

```diff
@@ -30,55 +36,202 @@ const addCardPopup = new PopupWithForm(
 
-// Form validators
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
-const addCardFormValidator = new FormValidator(config, addCardForm);
 profileEditFormValidator.enableValidation();
+const addCardFormValidator = new FormValidator(config, addCardForm);
 addCardFormValidator.enableValidation();
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
 
-// Event Listeners
-profileEditBtn.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.name;
-  profileDescriptionInput.value = userData.about;
-  profileEditPopup.open();
+document.addEventListener("DOMContentLoaded", () => {
+  const formElement = document.querySelector("#changeprofile_form");
+  const formValidator = new FormValidator(config, formElement);
+  formValidator.enableValidation();
 });
 
-addCardButton.addEventListener("click", () => addCardPopup.open());
+// Constants
+const profileEditBtn = document.querySelector("#profile__edit-button");
+const addCardButton = document.querySelector(".profile__add-button");
+const profileTitleInput = document.querySelector("#profile__name-input");
+const profileDescriptionInput = document.querySelector(
+  "#profile__subheading-input"
+);
+const editProfileImage = document.querySelector(".profile__avatar-edit");
 
-// Event handlers
+// Functions
 function handleProfileEditSubmit(profileData) {
-  const { modal__input_type_name: name, modal__input_type_description: about } =
-    profileData;
-  userInfo.setUserInfo(name, about);
-  profileEditPopup.close();
+  const name = profileData.modal__input_type_name;
+  const about = profileData.modal__input_type_description;
+
+  profileEditPopup.renderLoading(true);
+
+  return api
+    .patchProfileInfo(name, about)
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditForm.reset();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+    });
 }
 
 function handleAddCardSubmit(newCardData) {
-  renderer(newCardData);
-  addCardPopup.close();
+  addCardPopup.renderLoading(true);
+
+  return api
+    .postCards(newCardData)
+    .then((cardData) => {
+      renderer(cardData);
+      addCardPopup.close();
+      addCardForm.reset();
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
 }
 
-function handleImageClick(name, link) {
-  previewImagePopup.open({ name, link });
+function handleAvatarSubmit(formData) {
+  const avatarLink = formData["modal__form-input-link"];
+
+  editAvatarPopup.renderLoading(true);
+
+  return api
+    .patchProfileAvatar(avatarLink)
+    .then(() => {
+      userInfo.setAvatarPic(avatarLink);
+      editAvatarPopup.close();
+      avatarForm.reset();
```

Comment:
* Please, disable the button after the card submittion in `then` https://snipboard.io/IOWnzw.jpg 

* And after the avatar changing as well https://snipboard.io/Q8Pk6e.jpg 


```js

      avatarFormValidator.disableSubmitButton()

```